pkg/v2/resource/chart/update.go

```diff
@@ -2,14 +2,113 @@ package chart
 
 import (
 	"context"
+	"fmt"
+	"reflect"
 
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
+	"github.com/giantswarm/operatorkit/framework/context/resourcecanceledcontext"
+	"k8s.io/helm/pkg/helm"
+
+	"github.com/giantswarm/cluster-operator/pkg/v2/guestcluster"
 )
 
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
+	guestHelmClient, err := r.getGuestHelmClient(ctx, obj)
+	if guestcluster.IsNotFound(err) {
+		r.logger.LogCtx(ctx, "level", "debug", "message", "did not get a Helm client for the guest cluster")
+
+		// We can't continue without a Helm client. We will retry during the
+		// next execution.
```

Comment:
Would be cool to state why we can't and what is not found in this case. I can only guess now that it is the secrets?

Reply:
Exactly, the error we are checking here comes from https://github.com/giantswarm/cluster-operator/blob/master/pkg/v2/guestcluster/guestcluster.go#L120. I've updated the comment in create, update and delete to make clear that we are waiting for the cluster-operator cert for the current cluster.